Guides/First.md

```diff
@@ -0,0 +1,45 @@
+# First
```

Comment:
Since this API isn't `first`, but `firstNonNil`, can we align the naming of documents and headings accordingly?

Reply:
My thinking that it was a general category of functions for which others might be added. I'll amend it.